lib/IRGen/IRABIDetailsProvider.cpp

```diff
@@ -122,6 +126,32 @@ class IRABIDetailsProviderImpl {
     return {returnTy, {paramTy}};
   }
 
+  llvm::SmallVector<IRABIDetailsProvider::ABIAdditionalParam, 1>
+  getFunctionABIAdditionalParams(AbstractFunctionDecl *afd) {
+    llvm::SmallVector<IRABIDetailsProvider::ABIAdditionalParam, 1> params;
+    Mangle::ASTMangler mangler;
+
+    auto function = SILFunction::getFunction(SILDeclRef(afd), *silMod);
+    auto mangledName = mangler.mangleAnyDecl(afd, /*prefix=*/true);
+
+    if (function->getName() == mangledName) {
+      auto silFuncType = function->getLoweredFunctionType();
+      auto funcPointerKind =
+          FunctionPointerKind(FunctionPointerKind::BasicKind::Function);
+      auto signature =
+          Signature::getUncached(IGM, silFuncType, funcPointerKind);
+
+      for (unsigned i = 0; i < signature.getType()->getNumParams(); i++) {
+        auto set = signature.getAttributes().getAttributes(i);
+        if (set.hasAttribute(llvm::Attribute::AttrKind::SwiftSelf))
```

Comment:
What about the SwiftErrorResult it looks like you're only getting one parameter here?

Reply:
Yeah, the signature at this level doesn't have a `llvm::Attribute::AttrKind::SwiftError`, so we can only assure that the `SwiftSelf` is there